drivers/net/ethernet/intel/idpf/eth/idpf_eth.c

```diff
@@ -0,0 +1,380 @@
+
+	destroy_workqueue(adapter->post_init_wq);
+	destroy_workqueue(adapter->stats_wq);
+	mutex_destroy(&adapter->vport_ctrl_lock);
+
+	kfree(adapter);
+}
+
+/**
+ * idpf_eth_event_handler - Ethernet device event handler to proess events
+ * @dev_info: Device information
+ * @event: Event info
+ */
+static void idpf_eth_event_handler(struct idpf_eth_idc_dev_info *dev_info,
+				   struct idpf_eth_idc_event *event)
+{
+	struct idpf_eth_idc_auxiliary_dev *eth_dev;
+	struct idpf_eth_adapter *adapter;
+
+	eth_dev = container_of(dev_info, struct idpf_eth_idc_auxiliary_dev,
+			       eth_info);
+	adapter = dev_get_drvdata(&eth_dev->adev.dev);
+	if (!adapter)
+		return;
+
+	switch (event->event_code) {
+	case IDPF_ETH_IDC_EVENT_RESET_INITIATED:
+		set_bit(IDPF_ETH_RESET_IN_PROG, adapter->flags);
+		mutex_lock(&adapter->vport_ctrl_lock);
+		/* Avoid TX hangs on reset */
+		if (adapter->netdev)
+			idpf_netdev_stop(adapter->netdev);
+		idpf_set_vport_state(adapter);
+		mutex_unlock(&adapter->vport_ctrl_lock);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_RESET_COMPLETE:
+		clear_bit(IDPF_ETH_RESET_IN_PROG, adapter->flags);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_DEINIT:
+		mutex_lock(&adapter->vport_ctrl_lock);
+		idpf_eth_device_deinit(adapter);
+		mutex_unlock(&adapter->vport_ctrl_lock);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_LINK_CHANGE:
+		idpf_handle_event_link(adapter, event->event_data);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_REMOVE:
+		set_bit(IDPF_ETH_REMOVE_IN_PROG, adapter->flags);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_PROBE_POST_INIT:
+		adapter->dev_info->start_probe_post_init = true;
+		break;
+
+	default:
+		break;
+	}
+}
+
+/**
+ * idpf_eth_device_deinit - Ethernet device deinit routine
+ * @adapter: Ethernet Driver specific private structure
+ *
+ * Ethernet device initialization logic
+ */
+void idpf_eth_device_deinit(struct idpf_eth_adapter *adapter)
+{
+	cancel_delayed_work_sync(&adapter->post_init_task);
+	cancel_delayed_work_sync(&adapter->stats_task);
+	idpf_netdev_stop(adapter->netdev);
+	if (adapter->vport)
+		idpf_vport_dealloc(adapter->vport);
+}
+
+/**
+ * idpf_eth_get_reg_addr - reg operations
+ * @eth_shared: Ethernet shared information
+ * @reg_offset: Register offset
+ */
+void __iomem *idpf_eth_get_reg_addr(struct idpf_eth_shared *eth_shared,
+				    resource_size_t reg_offset)
+{
+	return (void __iomem *)(eth_shared->hw_addr + reg_offset);
+}
+
+static const struct auxiliary_device_id idpf_eth_id_table[] = {
+	{ .name = "idpf.eth", },
+	{ },
+};
+
+MODULE_DEVICE_TABLE(auxiliary, idpf_eth_id_table);
+
+static struct idpf_eth_idc_auxiliary_driver idpf_eth_driver = {
+	.adrv = {
+		.name = "eth",
+		.probe = idpf_eth_probe,
+		.remove = idpf_eth_remove,
+		.shutdown = idpf_eth_shutdown,
+		.id_table = idpf_eth_id_table
+	},
+	.event_handler = idpf_eth_event_handler
+};
+
+/**
+ * idpf_eth_get_driver - returns ethernet device driver structure
+ * @void: void
+ */
+struct idpf_eth_idc_auxiliary_driver *idpf_eth_get_driver(void)
```

Comment:
This func is not needed. Please check OOT PR. As per review comments, other means are used to get aux driver pointer in the main driver.

Reply:
Yea and the one version which was recommended in past reviews was to follow the RDMA /IDC model. So I changed it to that. Please check the OOT PR but I am pasting relevant snippet below.

static inline struct idpf_eth_auxiliary_driver *
idpf_eth_dev_to_driver(struct idpf_idc_eth_auxiliary_dev *eth_dev)
{
        return container_of(eth_dev->adev.dev.driver,
                            struct idpf_eth_auxiliary_driver,
                            adrv.driver);
}

Reply:
Please see my earlier comment. My version of implementation doesn't have argument to this function. It is void. So it is not possible for me to use container.